### PR TITLE
Add RuntimeService::block_runtime_code_heap_pages

### PR DESCRIPTION
**Pull request by tomaka**
cc https://github.com/paritytech/smoldot/issues/1769

This PR will fail CI because of an unused function warning, but it paves the way for #1769.

### CODE REVIEW

**Comment by tomaka** [comment]
Don't ask me why adding an unused method increases the size of the binary.

File: bin/light-base/src/runtime_service.rs [hunk 1]
```diff
@@ -495,6 +495,41 @@ impl<TPlat: Platform> RuntimeService<TPlat> {
         subscription.next().await.unwrap()
     }
 
+    /// Returns the value of the `:code` and `:heap_pages` storage items of the given block.
+    ///
+    /// Returns `None` if the runtime of that block isn't known by the runtime service, for
```
Comment (melekes) [inline bin/light-base/src/runtime_service.rs:500]: looks like this fn always returns `Some(...)`

File: bin/light-base/src/runtime_service.rs [hunk 2]
```diff
@@ -495,6 +495,41 @@ impl<TPlat: Platform> RuntimeService<TPlat> {
...
+                tree: Some(tree), ..
+            } => *tree
+                .input_iter_unordered()
+                .find(|block| block.user_data.hash == *block_hash)
+                .and_then(|block| block.async_op_user_data)?,
+            GuardedInner::FinalizedBlockRuntimeUnknown { tree: Some(tree) } => tree
+                .input_iter_unordered()
+                .find(|block| block.user_data.hash == *block_hash)
+                .and_then(|block| block.async_op_user_data)?
+                .unwrap(),
```
Comment (melekes) [inline bin/light-base/src/runtime_service.rs:525]: is it always safe to unwrap here?

File: bin/light-base/src/runtime_service.rs (hunk 1 above)
Comment (tomaka) [reply bin/light-base/src/runtime_service.rs:500]: There are two `?`s within the body of the function that can make it return `None`

Comment (melekes) [reply bin/light-base/src/runtime_service.rs:500]: Oh, right. Sorry

**Comment by tomaka** [comment]
I'm going to close this. It's only 40 lines of code with a merge conflict. While these lines of code may be useful later, it's better to clean up the list of PRs at this point.

**Pull request closed without merging**